test/logflare_web/controllers/api/source_controller_test.exs

```diff
@@ -0,0 +1,45 @@
+defmodule LogflareWeb.Api.SourceControllerTest do
+  use ExUnit.Case
+  use LogflareWeb.ConnCase
+
+  import Logflare.Factory
+  import Assertions
+
+  alias Logflare.Sources.Counters
+  alias Logflare.Teams
+
+  setup do
+    insert(:plan, name: "Free")
+    u1 = insert(:user)
+    u2 = insert(:user)
+    Teams.create_team(u1, %{name: "u1 team"})
+    Teams.create_team(u2, %{name: "u2 team"})
+
+    s1 = insert(:source, public_token: Faker.String.base64(16), user_id: u1.id)
+    s2 = insert(:source, user_id: u1.id)
+    s3 = insert(:source, user_id: u2.id)
+
+    Counters.start_link()
+
+    {:ok, users: [u1, u2], sources: [s1, s2, s3]}
+  end
+
+  describe "index/2" do
+    test "returns list of sources for given user", %{
+      conn: conn,
+      users: [u1, _u2],
+      sources: [s1, s2 | _]
+    } do
+      response =
+        conn
+        |> login_user(u1)
+        |> get("/api/sources")
+        |> json_response(200)
+
+      response = Enum.map(response, & &1["id"])
+      expected = Enum.map([s1, s2], & &1.id)
+
+      assert_lists_equal(response, expected)
```

Comment:
seems quite unnecessary to add `:assertions` for list comparison, esp when lists are so small. 
actually id matching would suffice

```
assert [%{"id"=> ^source_id}] = json_response(conn, 200)
```

Reply:
this was a bit of future-proofing since it's a lightweight dependency but can refactor to remove it. 👍